src/libAtomVM/opcodesswitch.h

```diff
@@ -422,6 +423,9 @@ typedef union
     }                                                                                               \
 }
 
+#define IS_EXTENDED_ALLOCATOR(code_chunk, base_index, off) \
+    ((code_chunk[(base_index) + (off)]) & 0xF) == COMPACT_EXTENDED
```

Comment:
There are several types of COMPACT_EXTENDED, I believe this test is only called with `COMPACT_EXTENDED_ALLOCATION_LIST` (0x37).

Reply:
yes. changed in [734d4b4](https://github.com/atomvm/AtomVM/pull/429/commits/734d4b4e81e03500ddb0d1903dfafc54c0dce457)